Add instructions for toolbar color customization (#58)
xournalpp/xournalpp.github.io: docs/guide/config/toolbar-colors.md

2. Restart Xournal++

### .gpl File Format

!!! note

    The `.gpl` file format is not authoritatively defined anywhere, so the file
    format described below may be specific to Xournal++. However, Xournal++
    tries to be compatible with `.gpl` files used with other software such as
    GIMP. If you find a `.gpl` file that works in other software but does _not_
    work with Xournal++, [please file a bug report](https://github.com/xournalpp/xournalpp/issues).

As an example, the default color palette of Xournal++ is shown below.

```
GIMP Palette
Name: Xournal Default Palette
#
0 0 0 Black
0 128 0 Green
0 192 255 Light Blue
0 255 0 Light Green
51 51 204 Blue
128 128 128 Gray
255 0 0 Red
255 0 255 Magenta
255 128 0 Orange
255 255 0 Yellow
255 255 255 White
```

If no existing `palette.gpl` is found, `palette.gpl` will be created with the
above contents.

The palette file consists of several parts:

 1. **Header**: every palette file must begin with the line `GIMP Palette`,
    otherwise the file will not be a valid `.gpl` file.
 2. A list of **attributes** of the form `Key: Value`, one per line. These
    attributes are metadata that describe the palette.

>  3. Lines beginning with `#` are comments, which are ignored by the file parser.

Reviewer: comments -> **comments**

Just to be consistent with the above highlights.

    The comment used above is to distinguish the attributes from the colors of
    the palette.

>  4. The colors of the palette, one per line, in the format `RRR GGG BBB Name`.

Reviewer: colors -> **colors**

Just to be consistent with the above highlights.

file.

The easiest way to get started with this is to let Xournal++ create the default palette file for you.
You can then use this as a template to adjust it to your needs.

Prerequisite: Xournal++ ran at least once with the new version

 1. Open `palette.gpl` in the [config file location](../file-locations.md)
 2. Edit the `palette.gpl` to your needs
 3. Restart Xournal++

### .gpl File Format

!!! note

    The `.gpl` file format is not authoritatively defined anywhere.
    Hence the below definition might only be valid for Xournal++.
    However, we are trying to be compatible with `.gpl` files as used in other Software.
    In case you find a case where the `.gpl` file is not parsable by Xournal++ but you think it should be parsable since for example GIMP accepts it, please file a bug report.

Lets start with an example:

```
GIMP Palette
Name: Xournal Default Palette
#
0 0 0 Black
0 128 0 Green
0 192 255 Light Blue
0 255 0 Light Green
51 51 204 Blue
128 128 128 Gray
255 0 0 Red
255 0 255 Magenta
255 128 0 Orange
255 255 0 Yellow
255 255 255 White
```

This is the default color palette of Xournal++, which is created if no palette is found.

> It consists of several Parts:

Reviewer 2: ```suggestion
It consists of several parts:
```

 1. Header: `GIMP Palette` any file not having exactly this string at the first line will not be accepted as `.gpl` file
 2. A list of attributes separated by `:`.
 3. An optional comment line as separator.
 4. The colors of the palette as space separated `RGB` values together with a name

The parser checks each line apart from the first line in the following order:

 1. Can line be parsed as **comment** (i.e. first character is `#`)
 2. Can line be parsed as **attribute** (i.e. two string separated by `:`)
 3. Can line be parsed as **color** (i.e. `RRR GGG BBB Name`)
 4. Otherwise abort parsing

In case Xournal++ cannot parse the palette file (i.e. it cannot parse a line), it loads the default palette.
If this happens there are two ways to get your palette file to load.

Option 1, Debugging:

 - run `xournalpp` from command line 
 - check the error message (it tells you which line was not parsable)
 - fix line in your `palette.gpl`

>  - retry

Reviewer 2: ```suggestion
 - Run `xournalpp` from command line 
 - Check the error message (it tells you which line was not parsable)
 - Fix line in your `palette.gpl`
 - Retry
```

Option 2, Fresh Start:

 - remove the `palette.gpl` file
 - start `xournalpp` (this will create a new default `palette.gpl` file)

>  - adjust `palett.gpl` to your needs

Reviewer 2: ```suggestion
 - Remove the `palette.gpl` file
 - Start `xournalpp` (this will create a new default `palette.gpl` file)
 - Adjust `palette.gpl` to your wishes
```

!!! note
    
    It's recommended to have at least 11 colors in your palette as the default toolbar has 11 ColorItems.

>     In case your palette has less, Xournal++ will show a warning in the CLI and just cycle through the palette.

Reviewer 2: ```suggestion
    In case your palette has fewer colors than the toolbar has ColorItems, Xournal++ will show a warning in the CLI and just cycle through the palette.
```
